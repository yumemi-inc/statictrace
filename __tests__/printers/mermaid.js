const { run } = require('../../build/lib');
const dotenv = require('dotenv');
const { MermaidPrinter } = require('../../build/printer');

dotenv.config();

it('Prints mermaid correctly', () => {
  const output = run(process.env.TS_PROJECT_TEST_CONFIG, new MermaidPrinter());
  expect(output).toMatchSnapshot();
});
