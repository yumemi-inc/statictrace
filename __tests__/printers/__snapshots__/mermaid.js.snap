// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prints mermaid correctly 1`] = `
"\`\`\`mermaid
graph TD
	Database_constructor --> get
	Database_constructor --> saveUser
	saveUser --> post
\`\`\`
\`\`\`mermaid
graph TD
	startRegistration --> processRegistration
	processRegistration --> someRegistrationProcedure
	someRegistrationProcedure --> callInsideTracedFn
	startRegistration --> finishRegistration
	finishRegistration --> callInsideTracedFn
	startRegistration --> cleanupSomething
\`\`\`
\`\`\`mermaid
graph TD
	begin --> funcA
	funcA --> funcC
	begin --> beingNestedEntrypoint
	beingNestedEntrypoint --> funcA
	beingNestedEntrypoint --> funcB
	begin --> funcB
\`\`\`
\`\`\`mermaid
graph TD
	beingNestedEntrypoint --> funcA
	funcA --> funcC
	beingNestedEntrypoint --> funcB
\`\`\`
\`\`\`mermaid
graph TD
	funcA --> funcC
\`\`\`
\`\`\`mermaid
graph TD
	infiniteSelfRecursion --> infiniteSelfRecursion
\`\`\`
\`\`\`mermaid
graph TD
	indirectInfiniteSelfRecursion --> makeIndirectRecursiveCall
	makeIndirectRecursiveCall --> indirectInfiniteSelfRecursion
\`\`\`"
`;
